etcdserver/server.go

```diff
@@ -272,6 +272,10 @@ func NewServer(cfg *ServerConfig) (*EtcdServer, error) {
 			var err error
 			str, err = discovery.JoinCluster(cfg.DiscoveryURL, cfg.DiscoveryProxy, m.ID, cfg.InitialPeerURLsMap.String())
 			if err != nil {
+				// It removes member directory when NewServer returns error.
+				// This prevents conflicts with 'proxy' directory when
```

Comment:
basically in the two !haveWAL case, we should clean up member dir if bootstrap fails.